File: Lista de exercicio/projetoJAVA/src/projetojava/questao6.java
package projetojava;

class Motor {
    int potencia;

    Motor(int potencia) {
        this.potencia = potencia;
    }
}

class Carro3 {
    String marca;
    String modelo;
    int ano;
    Motor motor;

    Carro3(String marca, String modelo, int ano, int potenciaMotor) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.motor = new Motor(potenciaMotor);
    }

    String detalhesCarro() {
        return marca + " " + modelo + " (" + ano + ") com motor de " + motor.potencia + " CV.";
    }
}

public class questao6 {
    public static void main(String[] args) {
        Carro3 carro1 = new Carro3("Toyota", "hilux", 2024, 204);
        Carro3 carro2 = new Carro3("Kya", "sorento", 2015, 194);
        Carro3 carro3 = new Carro3("Mitsubishi", "lancer", 2012, 160);

        System.out.println(carro1.detalhesCarro());
        System.out.println(carro2.detalhesCarro());
        System.out.println(carro3.detalhesCarro());
    }
}

File: Lista de exercicio/projetoPYTHON/questao6/questao6.py
class Motor:
    def __init__(self, potencia):
        self.potencia = potencia

class Carro:
    def __init__(self, marca, modelo, ano, potencia_motor):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.motor = Motor(potencia_motor)

    def detalhes_carro(self):
        return f"{self.marca} {self.modelo} ({self.ano}) com motor de {self.motor.potencia} CV."

def executar_questao6():
    carro1 = Carro("Toyota", "hilux", 2024, 204)
    carro2 = Carro("Kya", "sorento", 2015, 194)
    carro3 = Carro("Mitsubishi ", "lancer", 2012, 160)

    print(carro1.detalhes_carro())
    print(carro2.detalhes_carro())
    print(carro3.detalhes_carro())
